xarray/backends/scipy_.py

```diff
@@ -51,9 +51,13 @@ def __init__(self, variable_name, datastore):
         self.datastore = datastore
         self.variable_name = variable_name
         array = self.get_variable().data
-        self.shape = array.shape
```

Comment:
can you add a comment describing why this was necessary please?

Reply:
There was some discussion about this in #6239 as well.

Reply:
Ok, the abc inspection checks at a per-class level, so instance attributes cannot be resolved. 
I think there are two solutions to this:

1. Force subclasses to define `shape` as a property. Should be easy for internal classes, external ones might have to change or they stop working.
2. Don't use abstractmethod for shape and simply use `shape: tuple[int, ...]` in the Baseclass to tell type checkers that such a property should exist.

1) might be the cleaner solution since shape cannot be set anyway without breaking things.
Unfortunately it could break external libs.